Content.Client/_RMC14/NightVision/NightVisionOverlay.cs

```diff
@@ -41,23 +39,37 @@ protected override void Draw(in OverlayDrawArgs args)
         var eye = args.Viewport.Eye;
         var eyeRot = eye?.Rotation ?? default;
 
-        // TODO RMC14 this should use its own component
-        var entities = _entity.EntityQueryEnumerator<MobStateComponent, SpriteComponent, TransformComponent>();
-        while (entities.MoveNext(out var uid, out _, out var sprite, out var xform))
+        var entries = new List<NightVisionRenderEntry>();
```

Comment:
This should be turned into a readonly field that gets cleared at the start of Draw, for less allocations

Reply:
Done